### ISSUE TITLE
failed to lookup address information: Name or service not known

### ISSUE THREAD

**Issue by jzt308** [dani-garcia/bitwarden_rs#688]
I have a bitwarden docker instance running behind a traefik reverse proxy. I had no issues until a couple of bitwarden versions ago. Now I am unable to access the bitwarden site from several machines (on my home network) but on some (external) there is no issue.
The firefox plugin wont connect either but the app on my phone seems to sync.
The error/docker log is below the docker-compose: "failed to lookup address information: Name or service not known" while the browser displays an unexpected error has occured.
I've checked all the variables in the docker-compose and I googled the error but no luck in solving this yet. Any help would be appreciated.

Here is my docker-compose config:
version: '3.1'

services:

 bitwarden:
  image: mprasil/bitwarden:latest
  restart: unless-stopped
  volumes:
   - /DATA/app/bitwarden/data:/data/

  environment:
     SIGNUPS_ALLOWED: "true"
     INVITATIONS_ALLOWED: "true"
     ADMIN_TOKEN: "TOKENHERE"
     DOMAIN: "https://DOMAIN.HERE"
     SMTP_HOST: "internalmailhosthere"
     SMTP_FROM: "pass@DOMAIN.here"
     SMTP_PORT: "25"
     SMTP_SSL: "false"
     SMTP_USERNAME: "bitwarden@domain.here"
     SMTP_PASSWORD: "PASSWORDHERE"
     SHOW_PASSWORD_HINT: "false"
     LOG_FILE: "/data/bitwarden.log"
     LOG_LEVEL: "debug"
     
  networks:
   - web
   - default
     
  labels:
   - "traefik.docker.network=web"
   - "traefik.enable=true"
   - "traefik.backend=bitwarden"
   - "traefik.frontend.rule=Host:pass.domain.here"
   - "traefik.frontend.entryPoints=https"
      
networks:
  web:
    external: true

[2019-10-28 14:24:05][rocket::rocket][INFO] POST /api/accounts/prelogin application/json; charset=utf-8:
[2019-10-28 14:24:05][_][INFO] Matched: POST /api/accounts/prelogin (prelogin)
[2019-10-28 14:24:05][_][INFO] Outcome: Success
[2019-10-28 14:24:05][_][INFO] Response succeeded.
[2019-10-28 14:24:06][rocket::rocket][INFO] POST /identity/connect/token application/x-www-form-urlencoded; charset=utf-8:
[2019-10-28 14:24:06][_][INFO] Matched: POST /identity/connect/token (login)
thread '<unnamed>' panicked at 'called `Result::unwrap()` on an `Err` value: Io(Custom { kind: Other, error: "failed to lookup address information: Name or service not known" })', src/libcore/result.rs:1051:5

**Comment by dani-garcia** [comment]
Is your instance having troubles connecting to the SMTP server? Try disabling the SMTP settings for a bit and see if that helps.

**Comment by jzt308** [comment]
I just commented out all the lines regarding SMTP server but nio dice. Still the same error.

**Comment by BlackDex** [comment]
@jzt308 Seems to me like DNS Resolving is broken.
Are you able to resolve hostnames on the host where the docker container is running?
And also, try to check if you can resolve hosts from within the container if you can via the docker command.
```bash
docker exec -it <container-name> getent hosts <hostname-to-resolve>
# Example
docker exec -it bitwarden getent hosts github.com
```

**Comment by jzt308** [comment]
@BlackDex I am able to resolve the hosts:
140.82.114.4    github.com.

I just found out that the issue was me using my pi-Hole as a dns server. External is was able to login to bitwarden but from my internal network I wasn't able to. 
I changed the dns back to my firewall instead of the pi-hole and now it works!
So you where right the issue was dns. I will run some tests to figure out why my pi-hole sinks the connection from my lan.

Thnx!
